### PR TITLE
fix(catalog-msgraph): Encode query filters in requests made to msgraph

### LINKED ISSUE #30501: MS graph provider does not encode requests causing errors with some filters
### 📜 Issue Labels

- [x] Please familiarize yourself with the issue labels used in this project: [LABELS.md](https://github.com/backstage/backstage/blob/master/LABELS.md)

### 🔎 Search Terms

```plain
msgraph catalog provider encoding filter
```

### 🗃️ Project Area

Catalog

### 🔗 External Integration

Azure

### 📝 Description & Context

I'm using the Microsoft graph (entra id) catalog provider to ingest users. I want to have a filter to only ingest users from certain departments. See docs: https://backstage.io/docs/integrations/azure/org/#users 

For example:
```yaml
user:
  filter: department in('MARKETING', 'ENGINEERING & OPERATIONS')
```

This gives you an error: 
```
catalog error MicrosoftGraphOrgEntityProvider:default refresh failed, Error: Error while reading users from Microsoft Graph: BadRequest - Invalid filter clause: Found an unbalanced bracket expression.
```

However, a filter that does not have `&` is completely fine. For example:
```yaml
user:
  filter: department in('MARKETING', 'ENGINEERING AND OPERATIONS')
```

I traced the issue to this line of code where the encoding of the query in the API request to ms is set false (not encoding): https://github.com/backstage/backstage/blob/445cd3bb76db818f726d07fec998dae1b9050e10/plugins/catalog-backend-module-msgraph/src/microsoftGraph/client.ts#L199

I made a yarn patch to update the encoding from `false` to `true` and everything seems to work as expected and now I can use `&` in my filters. I notice in the code there is a note: "Microsoft Graph doesn't like an encoded query string". Is this still the case in 2025? I haven't noticed any issues with it set to true, so perhaps a simple PR can be made to switch this? Or if there is some other reason why it can't, then a config option should be helpful.

### 👍 Expected Behavior

I should be able to use '&' and other special characters in my filters and the MS graph provider should properly encode the string.

### 📦 Reproduction Repo

_No response_

### 🥾 Reproduction steps

Use a filter in MS graph provider that has a special character. For example:
```yaml
user:
  filter: department in('MARKETING', 'ENGINEERING & OPERATIONS')
```

### Have you read the Code of Conduct?

- [x] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes, and I have enough information to get started

### PR DESCRIPTION
**Pull request by stephenglass**
## Hey, I just made a Pull Request!

Fixes #30501 
Adds test case and tested in my org, but haven't sought out any "edge cases" (if there are any). As the changeset points out, if you are manually encoding characters in a filter you will need to update the filter to avoid being double encoded, this is why I made it a "minor" change.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by stephenglass** [comment]
not stale

**Comment by stephenglass** [comment]
Not stale

File: .changeset/slick-lamps-clap.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+'@backstage/plugin-catalog-backend-module-msgraph': minor
+---
+
+Encode query filters for requests made to msgraph. If you currently have manually encoded characters in a filter, this is a breaking change and must be updated to avoid requests being double encoded.
```
Comment (freben) [inline .changeset/slick-lamps-clap.md]:
```suggestion
**BREAKING**: Encode query filters for requests made to msgraph. If you currently have manually encoded characters in a filter, this is a breaking change and must be updated to avoid requests being double encoded.
```

this makes it easier to discover
